### PR TITLE
Add CI on AMD GPUs

### PR DESCRIPTION
**Pull request by pgrete**
## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] Docs build
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Comment by pgrete** [comment]
Alright, this is working and ready for review.
The new extended tests on AMD are not triggered yet (will only happens once the pipeline is in `develop`), but I confirmed directly on the machine that they run and pass.
Eventually,  I also want to move the Cuda and host tests to the new machine, but given that the old pipelines work it doesn't have a super high priority right now.

File: tst/regression/utils/test_case.py [hunk 1]
```diff
@@ -89,9 +89,7 @@ def __init__(self, run_test_path, **kwargs):
         try:
             parthenon_path = os.path.realpath(__file__)
             idx = parthenon_path.rindex("/parthenon/")
-            self.parameters.parthenon_path = os.path.join(
-                parthenon_path[:idx], "parthenon"
-            )
+            self.parameters.parthenon_path = parthenon_path[: idx + 10]
```
Comment (pgrete) [inline tst/regression/utils/test_case.py:92]: FYI this fixes an edge case where the path would be wrong if `parthenon` is checked out to `/parthenon`

File: cmake/TestSetup.cmake [hunk 2]
```diff
@@ -152,7 +152,7 @@ function(setup_test_parallel nproc dir arg extra_labels)
     list(APPEND labels "${extra_labels}")
 
     if(Kokkos_ENABLE_CUDA OR Kokkos_ENABLE_HIP)
-      set(PARTHENON_KOKKOS_TEST_ARGS "--kokkos-num-devices=${NUM_GPU_DEVICES_PER_NODE}")
+      set(PARTHENON_KOKKOS_TEST_ARGS "--kokkos-map-device-id-by=mpi_rank")
```
Comment (pgrete) [inline cmake/TestSetup.cmake:155]: This addresses https://github.com/parthenon-hpc-lab/parthenon/pull/934 to 0th order (not removing/cleaning up the `NUM_GPU_DEVICES_PER_NODE` var).

**Review by pdmullen — APPROVED** [review]
LGTM

File: cmake/TestSetup.cmake (hunk 2 above)
Comment (pdmullen) [reply cmake/TestSetup.cmake:155]: Can we close that one?

Comment (pgrete) [reply cmake/TestSetup.cmake:155]: Done, and further cleanup kept track in https://github.com/parthenon-hpc-lab/parthenon/issues/1133

**Review by BenWibking — APPROVED** [review]

**Review by lroberts36 — APPROVED** [review]

**Pull request merged**